Crash report "java.lang.RuntimeException: java.lang.InterruptedException"

UUID: 90fffa7a2844aa6bd2c9de4738f75e21
Build flavor: stable
Version name: v4.11.2
OS info: Android 9, sdk version: 28
Report description: No title
Logs:
```
=== STACKTRACE ===
Called from RxJava onError handler.
java.lang.RuntimeException: java.lang.InterruptedException
	at com.github.adamantcheese.chan.core.database.DatabaseManager.runTask(DatabaseManager.java:211)
	at com.github.adamantcheese.chan.core.manager.FilterEngine.getEnabledFilters(FilterEngine.java:122)
	at com.github.adamantcheese.chan.core.site.parser.ChanReaderRequest.<init>(ChanReaderRequest.java:98)
	at com.github.adamantcheese.chan.core.site.loader.ChanThreadLoader.getData(ChanThreadLoader.java:383)
	at com.github.adamantcheese.chan.core.site.loader.ChanThreadLoader.lambda$requestMoreDataInternal$1$ChanThreadLoader(ChanThreadLoader.java:256)
	at com.github.adamantcheese.chan.core.site.loader.-$$Lambda$ChanThreadLoader$qZBv59ha_WtrBFmKaX7Lq0W2Hrg.call(Unknown Source:2)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3666)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.InterruptedException
	at java.util.concurrent.FutureTask.awaitDone(FutureTask.java:420)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at com.github.adamantcheese.chan.core.database.DatabaseManager.runTask(DatabaseManager.java:209)
	... 16 more


=== SETTINGS ===
Prefetching enabled: false
Thread downloading enabled: true, active downloads = 0
Hi-res thumbnails enabled: false
Youtube titles parsing enabled: true
Youtube durations parsing enabled: false
Concurrent file loading chunks count: 2
WEBM streaming enabled: false
Saved files base dir info: Java API location: /storage/emulated/0/Kuroba/files, SAF location: , active: Java API
Local threads base dir info: Java API location: /storage/emulated/0/Kuroba/saved_threads, SAF location: , active: Java API
Phone layout mode: AUTO
```

--- a/Kuroba/app/src/main/java/com/github/adamantcheese/chan/Chan.java
+++ b/Kuroba/app/src/main/java/com/github/adamantcheese/chan/Chan.java
@@ -140,6 +140,11 @@ public class Chan
                 // fine, some blocking code was interrupted by a dispose call
                 return;
             }
+            if (e instanceof RuntimeException && e.getCause() instanceof InterruptedException) {
+                // fine, DB synchronous call (via runTask) was interrupted when a reactive stream
+                // was disposed of.
+                return;
+            }
             if (e instanceof FileCacheException.CancellationException
                     || e instanceof FileCacheException.FileNotFoundOnTheServerException) {
                 // fine, sometimes they get through all the checks but it doesn't really matter

--- a/Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/database/DatabaseManager.java
+++ b/Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/database/DatabaseManager.java
@@ -207,7 +207,12 @@ public class DatabaseManager {
     public <T> T runTask(final Callable<T> taskCallable) {
         try {
             return executeTask(taskCallable, null).get();
-        } catch (InterruptedException | ExecutionException e) {
+        } catch (InterruptedException e) {
+            // Since we don't rethrow InterruptedException we need to at least restore the
+            // "interrupted" flag.
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
             throw new RuntimeException(e);
         }
     }

